### DISCUSSION TITLE
Рассмотреть возможность использования Splade вместо Inertia

### DISCUSSION THREAD

**Discussion by Jampire** [moonshine-software/moonshine#240] [Ideas]
Привет

Кто что думает на счет использования [Laravel Splade](https://github.com/protonemedia/laravel-splade) вместо Inertia? Splade использует более нативный для laravel-разработчика стек (Blade + Vuejs), причем ковыряние во Vuejs практически не требуется. А сам Splade использует всё нативное из Laravel: не нужно переписывать контроллеры, достаточно переписать ресурсы и шаблоны.

Вкратце (более детально [тут](https://splade.dev/docs/splade-vs-inertia)):
- при использовании InertiaJs нужно всегда помнить по безопасность данных (без дополнительных усилий, всё, что передает контроллер, видно клиенту);
- В Splade почти всё рендерится на стороне сервера - более высокая безопасность;
- при использовании InertiaJs работа дублируется - все данные нужно дублировать на стороне клиента в `props`;
- при использовании InertiaJs требуются специальные методы на подобии `Inertia::render()`. Splade же использует нативный laravel-метод `view()`;
- не требуется раздутая команда разработки (бэкенд-разработчик/фронтэнд-разработчик/фуллстек). При использовании Splade достаточно бэкенд разработчиков (с минимальным знанием фронта).

**Comment by Sanes** [comment]
Можно еще сильнее упростить. Мне Hotwire Turbo отлично зашел. Есть пакет для Laravel. 
В Laravel Orchid используется и вроде не жалуются.

**Reply by Jampire** [reply]
Глубоко Hotwire не копал, но то, что лежит на поверхности в документации кажется не упрощением, а усложнением по сравнению с Splade:
- Hotwire требует корректный маппинг DOM-индексов между шаблонами в пределах одной функциональности (лишняя работа, источников багов);
- контроллеры не возвращают `view()`, а возвращают `turbo_stream()` (кастомная реализация, лишняя работа по переписыванию контроллеров);
- Hotwire для серьезных вещей требует написания js-контроллеров;
- Hotwire использует ajax-запросы для обновления данных. Проблемы тут: задержка по сети и размер данных в запросе;
- Splade использует Vue engine для двустроннего биндинга (отсутствуют ajax-запросы?).

**Reply by Sanes** [reply]
В Splade вы не уйдете от Vue компонентов. Вряд ли там все потребности закрыты. В Turbo классический Blade.
> контроллеры не возвращают view()

Возвращают

**Reply by Jampire** [reply]
Splade это тоже классический Blade (отражено даже в названии) и в стадии активной разработки. Но уже сейчас всё достаточно хорошо закрыто от прямого использования Vue.  Hotwire на рынке 3 года (?) и всё равно местами требуется использование js-контроллеров.

Под возвратом `view` я подразумевал, что в случае Spade, файл контроллера вообще открывать не нужно. А вот в случае с Hotwire в каждый метод контроллера, который должен работать с Hotwire, нужно будет дописать:
```php
 if ($request->wantsTurboStream()) {
    return turbo_stream(...);
}
```
Возможно, у вас как пользователя этой технологии, есть обходные пути решения этой проблемы. Я же сужу по официальной документации компонента под Laravel. Т.е. тут как минимум:
- лишняя работа по переписыванию контроллеров - потенциальный источник багов;
- подобный подход с использованием условий попахивает (code smell) - нарушение принципов SOLID.

Что я вижу еще в документации по Hotwire: требуется использование Broadcasting для полноценной работы. В документации же к Splade такого требования нет. Нужно, конечно проверить дополнительно Splade. Но если Hotwire действительно требует broadcasting, то это лишняя работа по установке/настройке broadcasting-сервера. Плюс дополнительные деньги для конечных пользователей продукта по содержанию broadcasting-сервера на стороне сервис провайдера.

Даже чисто с маркетинговой точки зрения: если на рынке есть два одинаковых решения, я бы не стал использовать решение конкурентов (Orchid + Hotwire).

**Comment by lee-to** [comment]
Мы пока не уверенны что выберем inertia, в целом и текущий стек хорош
